src/Native/System.Security.Cryptography.Native/pal_ecdsa_import_export.cpp

```diff
@@ -0,0 +1,461 @@
+	BIGNUM** B, int32_t* cbB,
+	BIGNUM** Gx, int32_t* cbGx,
+	BIGNUM** Gy, int32_t* cbGy,
+	BIGNUM** n, int32_t* cbn,
+	BIGNUM** h, int32_t* cbh,
+	BIGNUM** S, int32_t* cbS)
+{
+	// get the public key parameters
+	int32_t rc = CryptoNative_GetECKeyParameters(publicKey, includePrivate, qx, cbQx, qy, cbQy, d, cbd);
+
+	if (!P || !cbP ||
+		!A || !cbA ||
+		!B || !cbB ||
+		!Gx || !cbGx ||
+		!Gy || !cbGy ||
+		!n || !cbn ||
+		!h || !cbh ||
+		!S || !cbS)
+	{
+		assert(false);
+		// since these parameters are 'out' parameters in managed code, ensure they are initialized
+		if (P) *P = nullptr; if (cbP) *cbP = 0;
+		if (A) *A = nullptr; if (cbA) *cbA = 0;
+		if (B) *B = nullptr; if (cbB) *cbB = 0;
+		if (Gx) *Gx = nullptr; if (cbGx) *cbGx = 0;
+		if (Gy) *Gy = nullptr; if (cbGy) *cbGy = 0;
+		if (n) *n = nullptr; if (cbn) *cbn = 0;
+		if (h) *h = nullptr; if (cbh) *cbh = 0;
+		if (S) *S = nullptr; if (cbS) *cbS = 0;
+
+		return 0;
+	}
+
+	// optional parameters
+	*S = nullptr;
+	*cbS = 0;
+
+	EC_GROUP* group = nullptr;
+	EC_POINT* G = nullptr;
+	EC_METHOD* curveMethod = nullptr;
+	BIGNUM* xBn = nullptr;
+	BIGNUM* yBn = nullptr;
+	BIGNUM* pBn = nullptr;
+	BIGNUM* aBn = nullptr;
+	BIGNUM* bBn = nullptr;
+	BIGNUM* nBn = nullptr;
+	BIGNUM* hBn = nullptr;
+	BIGNUM* sBn = nullptr;
+
+	// exit if CryptoNative_GetECKeyParameters failed
+	if (rc != 1) goto error;
+
+	xBn = BN_new();
+	yBn = BN_new();
+	pBn = BN_new();
+	aBn = BN_new();
+	bBn = BN_new();
+	nBn = BN_new();
+	hBn = BN_new();
+
+	if (!xBn || !yBn || !pBn || !aBn || !bBn || !nBn) goto error;
+
+	group = const_cast<EC_GROUP*>(EC_KEY_get0_group(publicKey)); // curve
+	if (!group) goto error;
+
+	curveMethod = const_cast<EC_METHOD*>(EC_GROUP_method_of(group));
+	if (curveMethod == nullptr) 
+        goto error;
+
+	*curveType = MethodToCurveType(curveMethod);
+	if (*curveType == ECCurveType::Unspecified) 
+        goto error;
+
+	// extract P, A, B
+	if (*curveType == ECCurveType::Characteristic2)
+	{
+		if (!EC_GROUP_get_curve_GF2m(group, pBn, aBn, bBn, nullptr)) 
+            goto error;
+	}
+	else
+	{
+		if (!EC_GROUP_get_curve_GFp(group, pBn, aBn, bBn, nullptr)) 
+            goto error;
+	}
+
+	// extract Gx and Gy
+	G = const_cast<EC_POINT*>(EC_GROUP_get0_generator(group));
+	if (*curveType == ECCurveType::Characteristic2)
+	{
+		if (!EC_POINT_get_affine_coordinates_GF2m(group, G, xBn, yBn, NULL)) 
+            goto error;
+	}
+	else
+	{
+		if (!EC_POINT_get_affine_coordinates_GFp(group, G, xBn, yBn, NULL)) 
+            goto error;
+	}
+
+	// extract n
+	if (!EC_GROUP_get_order(group, nBn, nullptr)) 
+        goto error;
+
+	// extract h
+	if (!EC_GROUP_get_cofactor(group, hBn, nullptr)) 
+        goto error;
+
+	// extract S (optional)
+	if (EC_GROUP_get0_seed(group))
+	{
+		sBn = BN_bin2bn(EC_GROUP_get0_seed(group), 
+			static_cast<int>(EC_GROUP_get_seed_len(group)), NULL);
+		// todo: determine how to also extract the nid to the algorithm (eg NID_sha256) that was used with seed
```

Comment:
I agree but only if it is actually possible. I don't think there is actually metadata for that -- see \crypto\ec\ec_curve.c and there is no metadata for the algorithm -- just the seed. We could provide a hard-coded mapping, but that could be done in managed code. For now, I'll update the comment based on what I know